openlibrary/plugins/openlibrary/js/edit.js

```diff
@@ -70,30 +70,163 @@ export function initRoleValidation() {
     });
 }
 
-export function initIdentifierValidation() {
+/**
+ * isIsbnDedupe takes an isbn string and returns true if the given ISBN
+ * is already added to this edition.
+ * @params isbn - ISBN string duplication checking
+ * @returns true if the given ISBN is already added to the edition
+ */
+function isIsbnDupe(isbn) {
+    const isbnEntries = document.querySelectorAll('.isbn_10, .isbn_13');
+    for (let i = 0; i < isbnEntries.length; i++) {
+        if (isbn === isbnEntries[i]['value']) {
+            return true;
+        }
+    }
+    return false;
+}
+
+/**
+ * isChecksumValidIsbn10 checks the format and validation for ISBN 10.
+ * Adapted from https://www.oreilly.com/library/view/regular-expressions-cookbook/9781449327453/ch04s13.html
+ * @params isbn - ISBN string for validating
+ * @returns true if ISBN string is a valid ISBN 10
+ */
+export function isChecksumValidIsbn10(isbn) {
```

Comment:
We could import books with isbns that don't have a valid checksum.
It's known that books sometimes are published with "invalid" isbns https://en-academic.com/dic.nsf/enwiki/8948#cite_ref-18

But I imagine we still want to import them.

So, I'm not sure how/if we should use it. Perhaps it could be a warning ("this looks like an invalid isbn, are you sure you want to continue?")?

Reply:
Oh, thank you for catching this. I will see what I can do.